### PR TITLE
More work on the way to assembly generation from Kamelets

### PR DESCRIPTION
**Pull request by fhcarter**
(part of bigger issue)

Correct generation of assembly projects so that routes are correct, naming is correct, and the things work from generation to install & operation (for one simple case).

Added Camel version used into package names & assembly name.  This will allow us to support > 1 camel version in our catalog (once we do that).  Kamelets are Camel version specific -- there's an issue in the current state because our connector & component predate kamelets, and there was a change to (at least) the logging component in the minor versions in between.  Will correct that versioning overall in a subsequent PR.  Separate PR up to unify the versioning, so it'll be easier to change.

### CODE REVIEW

**Comment by fhcarter** [comment]
[Jenkins run](https://jenkins.vantiq.com/jenkins/job/vantiq-extension-sources-branch/144/)

File: buildSrc/src/main/groovy/io/vantiq/extsrc/assy/tasks/AssemblyResourceGeneration.groovy [hunk 1]
```diff
@@ -51,10 +60,18 @@ class AssemblyResourceGeneration extends DefaultTask {
     public static final String KAMELET_SOURCE = 'kamelet:source'
     public static final String KAMELET_SINK = 'kamelet:sink'
     
-    public static final String ASSEMBLY_PACKAGE_BASE = 'io.vantiq.extsrc.camel.kamelets'
```
Comment (fhcarter) [inline buildSrc/src/main/groovy/io/vantiq/extsrc/assy/tasks/AssemblyResourceGeneration.groovy:54]: naming to keep vantiq installation & rules happy

File: buildSrc/src/main/groovy/io/vantiq/extsrc/assy/tasks/AssemblyResourceGeneration.groovy [hunk 2]
```diff
@@ -67,19 +84,20 @@ class AssemblyResourceGeneration extends DefaultTask {
     public static final String SERVICE_NAME_SUFFIX = '_service'
     public static final String SOURCE_NAME_SUFFIX = '_source'
 
-    public static final String CAMEL_CONNECTOR_SOURCE_TYPE = 'CAMEL_SOURCE'
```
Comment (fhcarter) [inline buildSrc/src/main/groovy/io/vantiq/extsrc/assy/tasks/AssemblyResourceGeneration.groovy:70]: Move to def'n more in line with Vantiq naming

File: buildSrc/src/main/groovy/io/vantiq/extsrc/assy/tasks/AssemblyResourceGeneration.groovy [hunk 3]
```diff
@@ -144,6 +162,11 @@ class AssemblyResourceGeneration extends DefaultTask {
         kameletJarFilename = kameletJarFilename.substring(0, kameletJarFilename.indexOf('!/'))
         kameletJarFilename = kameletJarFilename.substring(KAMELETS_RESOURCE_PATH.length())
```
Comment (fhcarter) [inline buildSrc/src/main/groovy/io/vantiq/extsrc/assy/tasks/AssemblyResourceGeneration.groovy:164]: Capture version from jar artifact.  Transform it to be a valid package segment/vantiq entity name by replacing periods with underscores.

File: buildSrc/src/main/groovy/io/vantiq/extsrc/assy/tasks/AssemblyResourceGeneration.groovy [hunk 4]
```diff
@@ -227,79 +250,83 @@ class AssemblyResourceGeneration extends DefaultTask {
...
+            props.forEach((k, v) -> {
+                //noinspection unchecked
+                log.debug('        name: {}, [type: {}, desc: {}]', k,
+                    ((Map<String, String>) v).get('type'),
+                    ((Map<String, String>) v).get('description'))
+            })
+        }
         String kamName = kameletDefinition.substring(0, kameletDefinition.indexOf('.kamelet.yaml'))
         kamName = kamName.replace('-', '_')
-        String packageName = String.join(PACKAGE_SEPARATOR, ASSEMBLY_PACKAGE_BASE, kamName)
```
Comment (fhcarter) [inline buildSrc/src/main/groovy/io/vantiq/extsrc/assy/tasks/AssemblyResourceGeneration.groovy:239]: include camel version in package name

File: buildSrc/src/main/groovy/io/vantiq/extsrc/assy/tasks/AssemblyResourceGeneration.groovy [hunk 5]
```diff
@@ -432,43 +474,31 @@ class AssemblyResourceGeneration extends DefaultTask {
...
-        log.info('Found projects source reference: {}', sourceName)
-        sourceName = sourceName.substring(sourceName.lastIndexOf('/') + 1)
-        log.info('Found projects source name {}', sourceName)
 
         project.options = [
             description: description,
             filterBitArray: 'ffffffffffffffffffffffffffffffff',
             type: 'dev',
             v: 5,
             isModeloProject: true,
```
Comment (fhcarter) [inline buildSrc/src/main/groovy/io/vantiq/extsrc/assy/tasks/AssemblyResourceGeneration.groovy:483]: We create assemblies that are _layered on_ (have a dependency on or prerequisite of) the connector def'n assembly that includes the sourceimpl & schema type.  We don't want to delete those when this assembly is uninstalled, so exclude things things are that are added in via dependency processing.

File: buildSrc/src/main/groovy/io/vantiq/extsrc/assy/tasks/AssemblyResourceGeneration.groovy [hunk 6]
```diff
@@ -377,22 +403,37 @@ class AssemblyResourceGeneration extends DefaultTask {
...
                               (CAMEL_RUNTIME_ROUTE_DOCUMENT): routeDoc]
-        def propValStubs = [:]
+        Map<String, String> propValStubs = [:]
+        List<String> propNameList = []
+
+        // Note: Here, we could have the CAMEL source type define camelRuntime,propertyValues as configurable.
+        // However, if we did that, on installation, the user would have to provide an object with the various lower
+        // level items & their values.  And any descriptive information provided by the kamelet definition would be
+        // lost.  Consequently, it seems preferable for each source to define the set it needs and let the assembly
+        // installation and configuration mechanism take it's course.
```
Comment (rblumer4) [inline buildSrc/src/main/groovy/io/vantiq/extsrc/assy/tasks/AssemblyResourceGeneration.groovy:418]: typo: it's -> its

**Review by jcguion — APPROVED** [review]
LGTM

**Pull request merged**